components/brave_wallet_ui/common/hooks/swap.ts

```diff
@@ -410,13 +410,15 @@ export default function useSwap (
     }
 
     if (swapValidationError === 'insufficientAllowance' && allowance) {
-      const fromAssetBalanceWeiHex = toWeiHex(fromAssetBalance, fromAsset.asset.decimals)
+      const allowanceHex = !fromAssetBalance
+        ? toWeiHex(fromAmount, fromAsset.asset.decimals)
+        : toHex(fromAssetBalance)
```

Comment:
If we're unable to fetch the balance (ex, if asset is in swap options list, but not in visible options list), we set the allowance to the value of the swap transaction.